python/graphstorm/gconstruct/utils.py

```diff
@@ -530,6 +538,107 @@ def to_tensor(self):
         """
         return th.tensor(self.to_numpy())
 
+class ExtFeatureWrapper(ExtNumpyWrapper):
+    """ The wrapper to memory-mapped Numpy array when combining features
+
+    Parameters
+    ----------
+    arr_path : str
+        The path to different feature files directory
+    shape : tuple
+        The shape of the array.
+    dtype : numpy dtype
+        The data type.
+    merged_file: str
+        The merged file name
+    """
+    def __init__(self, arr_path, shape=None, dtype=None, merged_file="merged_feature.npy"):
+        super().__init__(arr_path, shape, dtype)
+        self._directory_path = arr_path
+        self._merged_file = merged_file
+        self._wrapper = []
+        self._arr_path = os.path.join(self._directory_path, self._merged_file)
+
+    def __getitem__(self, idx):
+        if not self._shape:
+            raise RuntimeError("Call ExtFeatureWrapper.merge() first before calling __getitem__")
+        if self._arr is None:
+            self._arr = np.memmap(self._arr_path, self._dtype, mode="r",
+                                  shape=self._shape)
+        return self._arr[idx].astype(self.dtype)
+
+    def __del__(self):
+        self.cleanup()
```

Comment:
Can you add a unitest to check if __del__ works in the unit test?

Reply:
It is already there, checkout https://github.com/awslabs/graphstorm/blob/7d140e150d1fc3c3ef12fbaa9b55e17d0915b429/tests/unit-tests/gconstruct/test_construct_graph.py#L1686.